app/models/manageiq/providers/vmware/cloud_manager/vm/reconfigure.rb

```diff
@@ -28,6 +28,11 @@ def disk_default_type
     'LSI Logic Parallel SCSI'
   end
 
+  def available_adapter_names
+    available = (0..3).to_a - network_ports.map { |nic| nic_idx(nic.name) }.map(&:to_i)
```

Comment:
Just want to leave a comment here for the record that we believe 4 NICs is a current UI limitation, if this changes in the future we should come back to this.